packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -84,9 +84,15 @@ function popProvider(providerFiber: Fiber): void {
   if (isPrimaryRenderer) {
     context._currentValue = currentValue;
     context._changedBits = changedBits;
+    if (__DEV__) {
+      context._currentRenderer = null;
```

Comment:
I'm not very familiar with this code but I think it would be safer if we did this conditionally only when `context._currentRenderer === rendererSigil`.